### DISCUSSION TITLE
Importacion de Modulos

### DISCUSSION THREAD

**Discussion by cdv13** [santiemanuel/milProgramadoresQA#30] [Preguntas y Respuestas] [answered]
Hola Santi,
Estaba teniendo problemas para importar un modulo que lo tengo en otra carpeta, probé las siguientes formas, pero ninguna funciono:

`from ProyectoFinal.Clases import producto`
`from Clases.producto import Producto`

Las carpetas estarían organizadas así:

1. ProyectoFinal
    - Clases
      - producto.py
    - Grafica
      - cargaproducto.py (aquí quiero importar)

**Comment by santiemanuel** [comment, marked as answer]
Hola Clara, es algo complicado pero te paso el código. Supongamos que las carpetas se llaman `carpetaUno` y `carpetaDos`. Quiero usar una función en `carpetaUno` en un módulo de `carpetaDos`, entonces aquí hago:

- Archivo modulo_uno.py en carpetaUno

```py
def mostrar_suma(a, b):
    print(f"La suma es: {a+b}")
```

- Archivo modulo_dos.py en carpetaDos

```py
from pathlib import Path
import sys

sys.path.append(str(Path(__file__).parent.parent))
import carpetaUno.modulo_uno as modulo_uno

def usando_afuera(a, b, nombre):
    modulo_uno.mostrar_suma(a, b)
    print(f"El nombre es: {nombre}")

if __name__ == "__main__":
    usando_afuera(2, 7, "Santiago")

**Reply by cdv13** [reply]
Gracias Santi!!! no daba con una posible solución, ahi lo voy a probar a ver como me funciona
